docs/docs/transforms/label.md

```diff
@@ -0,0 +1,123 @@
+| avoidMarks    | {% include type t="String[]" %} | The property `avoidMarks` is useful when you want to take other objects in the chart into consideration when placing label. For example, in connected scatter plot in below example, we do not want label to the data point itself, but we also do not want a line to cross over our labels. So, we have to add the name of the line mark into `avoidMarks` to avoid the line when placing labels. **Default value:** `[]` |
+| avoidBaseMark | {% include type t="Boolean" %}  | the flag to specify if you want you labels to collide with the base mark. This flag is usually `true` (no allowing collision between labels and base mark); however, in some case like area chart, we want to place each label inside its area, so this flag has to be `false`. **Default value:** `true`                                                                                                                  |
+| lineAnchor    |  {% include type t="String" %}  | When labeling group-line mark, you can only have one label per 1 line. `lineAnchor` specify if you want the labels to be at the beginning (`"begin"`) or at the end (`"end"`) of the line. **Note**: this property only works with group line mark as the base mark. **Default value:** `"end"`                                                                                                                            |
+| markIndex     |  {% include type t="Number" %}  | `markIndex` is used to specify which mark in the group mark you want to label because group mark can have more than 1 mark in the group. Regularly, `markIndex` is used with group-line and group-area mark. **Note**: `markIndex` only works with group mark as the base mark. **Default value:** `0`                                                                                                                     |
+| as            | {% include type t="String[]" %} | The output fields written by the transform. The default is `['x', 'y', 'opacity', 'align', 'baseline', 'originalOpacity', 'transformed']`                                                                                                                                                                                                                                                                                  |
+
+## Usage
+
+### <a name="concept"></a>Basic concept
+
+{% include embed spec="label-examples/basic-concept-no-label" %}
+
+In this scatter plot example, we can add label to each point by adding text mark with the same `x` and `y` encoding channels to the chart. Since `x` and `y` encoding channels of the text mark follow the ones from the symbol mark (as we call this base mark), we can use reactive geometry to get information from the symbol mark and only modify the text.
```

Comment:
```suggestion
In this scatter plot example, we can add labels to each point by adding a text mark with the same `x` and `y` encoding channels to the chart. Since `x` and `y` encoding channels of the text mark follow the ones from the symbol mark (as we call this base mark), we can use reactive geometry to get information from the symbol mark and only modify the text.
```